docs/USING_PRO.md

```diff
@@ -261,6 +261,8 @@ Hooks are methods that hook into some part of marked. The following hooks are av
 | `preprocess(markdown: string): string` | Process markdown before sending it to marked. |
 | `postprocess(html: string): string` | Process html after marked has finished parsing. |
 | `processAllTokens(tokens: Token[]): Token[]` | Process all tokens before walk tokens. |
+| `provideLexer(): (src: string, options?: MarkedOptions) => Token[]` | Provide function to tokenize markdown. |
+| `provideParser(): (tokens: Token[], options?: MarkedOptions) => Token[]` | Provide function to parse tokens. |
```

Comment:
The output of a parser should be a string, no?

Reply:
Could we add an example showing how these might be used for chunking? For example preserving the `links` between renders? Or add it to the tests?

Reply:
Nice catch! 

The default return value is a string but really it can be anything. The output of this function is returned as the return value for `marked.parse` so if someone wanted to create a new parser to parse the markdown into react elements like [marked-react](https://www.npmjs.com/package/marked-react) They could return a react element.